src/validation/custom.validation.ts

```diff
@@ -151,25 +152,55 @@ const checkStartDateFields = (dayStr: string = "", monthStr: string = "", yearSt
   }
 };
 
-export const checkDateFieldDay = (dayMissingMessage: string, dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+export const checkDateFieldDay = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
   if (isYearEitherMissingOrCorrectLength(yearStr)) {
     if (dayStr === "" && monthStr !== "" && yearStr !== "") {
-      throw new Error(dayMissingMessage);
+      throw new Error(ErrorMessages.DAY);
     } else if (dayStr === "" && monthStr === "" && yearStr !== "") {
       throw new Error(ErrorMessages.DAY_AND_MONTH);
     } else if (dayStr === "" && monthStr !== "" && yearStr === "") {
       throw new Error(ErrorMessages.DAY_AND_YEAR);
+    } else {
+      checkDateIsNotCompletelyEmptyWithErrorThrown(ErrorMessages.ENTER_DATE, dayStr, monthStr, yearStr);
     }
   }
   return true;
 };
 
-export const checkDateFieldMonth = (monthMissingMessage: string, dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+export const checkDateFieldDayOfBirth = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  if (isYearEitherMissingOrCorrectLength(yearStr)) {
+    if (dayStr === "" && monthStr !== "" && yearStr !== "") {
+      throw new Error(ErrorMessages.DAY_OF_BIRTH);
+    } else if (dayStr === "" && monthStr === "" && yearStr !== "") {
+      throw new Error(ErrorMessages.DAY_AND_MONTH_OF_BIRTH);
+    } else if (dayStr === "" && monthStr !== "" && yearStr === "") {
+      throw new Error(ErrorMessages.DAY_AND_YEAR_OF_BIRTH);
+    } else {
+      checkDateIsNotCompletelyEmptyWithErrorThrown(ErrorMessages.ENTER_DATE_OF_BIRTH, dayStr, monthStr, yearStr);
```

Comment:
See comment above. So here you'd have to check for 'false' being returned and if so throw the error...

Reply:
Removed second function